src/decanter/core/core_api/train_input.py

```diff
@@ -100,90 +100,93 @@ def get_train_params(self):
 class TrainTSInput:
     """Train Input for ExperimentTS Job.
 
-    Settings for time series training.
+    Settings for auto time series forecast training.
 
     Attributes:
         data (:class:`~decanter.core.jobs.data_upload.DataUpload`): Train data uploaded on
             Decanter Core server
-        train_body (:class:`~decanter.core.extra.body_obj.TrainTrainAutoTSBodyBody`):
+        train_body (:class:`~decanter.core.core_api.body_obj.TrainAutoTSBody`):
             Request body for sending time series training api.
     """
     def __init__(
-            self, data, target, datetime_column, endogenous_features,
-            forecast_horizon, gap, time_unit, regression_method,
-            classification_method, callback=None, max_iteration=None,
-            max_window_for_feature_derivation=None, generation_size=None,
-            mutation_rate=None, crossover_rate=None, tolerance=None,
-            validation_percentage=None, max_model=None, seed=None,
-            evaluator=None, max_run_time=None, apu=None, algos=None,
-            nfold=None, balance_class=None, max_after_balance=None,
-            sampling_factors=None, test_base_id=None, holdout_percentage=None,
-            time_group=None, feature_types=None):
+        self, data, target, datetime_column, forecast_horizon, gap, id=None, data_type=None,
+        callback=None, version='v2', max_iteration=None, generation_size=None, mutation_rate=None,
+        crossover_rate=None, tolerance=None, validation_percentage=None, holdout_percentage=None,
+        max_model=None, seed=None, evaluator=None, max_run_time=None, nfold=None,time_unit=None,
+        numerical_groupby_method=None, categorical_groupby_method=None,endogenous_features=None,
+        exogenous_features=None, nullable=None, time_groups=None, max_window_for_feature_derivation=None):
 
         self.data = data
-        model_build_control = CoreBody.ModelBuildControl.create(
-            tolerance=tolerance,
-            validation_percentage=validation_percentage,
-            max_model=max_model)
 
-        genetic_algorithm = CoreBody.GeneticAlgorithmParams.create(
+        if id is None:
+            columns = None
+        elif nullable != None:
+            if len(id) == len(data_type) and len(data_type) == len(nullable) :
+                columns = map(
+                        lambda tuple3:CoreBody.Column.create(id=tuple3[0], data_type=tuple3[1], nullable=tuple3[2]),
```

Comment:
tuple3 is a wired variable name.
Is it possible to unpack it with `lambda id, data_type, nullable: ...., id`?

Reply:
What is this nullable? I can't seem to find it in CoreX API

Reply:
@schwannden
The nullable field is in the requestBody of `/v2/tasks/setup` -> `SetupBody` -> `data_columns*: [Column]` -> `(id*, data_type*, nullable)`